commons/src/main/java/com/powsybl/commons/config/PlatformConfigNamedProvider.java

```diff
@@ -0,0 +1,197 @@
+         * {@code moduleName} in {@code platformConfig} among the {@code providers}
+         * arguments based on its name.
+         *
+         * @return the provider
+         */
+        public static <T extends PlatformConfigNamedProvider> T findDefault(String moduleName,
+                Class<T> clazz, PlatformConfig platformConfig) {
+            return find(null, moduleName,
+                    Arrays.asList(DEFAULT_SERVICE_IMPL_NAME_PROPERTY), clazz,
+                    platformConfig);
+        }
+
+        /**
+         * Find the provider among the {@code providers} based on its {@code name}, or
+         * if {@code name} is null find the default provider like @{link findDefault}
+         *
+         * @return the provider
+         */
+        public static <T extends PlatformConfigNamedProvider> T find(String name, String moduleName,
+                Class<T> clazz, PlatformConfig platformConfig) {
+            return find(name, moduleName,
+                    Arrays.asList(DEFAULT_SERVICE_IMPL_NAME_PROPERTY), clazz,
+                    platformConfig);
+        }
+
+        /**
+         * Find the default provider configured in the standard field or the legacy
+         * field of {@code moduleName} in {@code platformConfig} among the
+         * {@code providers} arguments based on its name.
+         *
+         * @deprecated Use {@link #findDefault} instead
+         *
+         * @return the provider
+         */
+        @Deprecated
+        public static <T extends PlatformConfigNamedProvider> T findDefaultBackwardsCompatible(
+                String moduleName, Class<T> clazz, PlatformConfig platformConfig) {
+            return find(null, moduleName,
+                    Arrays.asList(DEFAULT_SERVICE_IMPL_NAME_PROPERTY,
+                    LEGACY_SERVICE_IMPL_NAME_PROPERTY), clazz, platformConfig);
+        }
+
+        /**
+         * Find the provider among the {@code providers} based on its {@code name}, or
+         * if {@code name} is null find the default provider like @{link
+         * findDefaultBackwardsCompatible}
+         *
+         * @deprecated Use {@link #find} instead
+         *
+         * @return the provider
+         */
+        @Deprecated
+        public static <T extends PlatformConfigNamedProvider> T findBackwardsCompatible(String name,
+                String moduleName, Class<T> clazz, PlatformConfig platformConfig) {
+            return find(name, moduleName,
+                    Arrays.asList(DEFAULT_SERVICE_IMPL_NAME_PROPERTY,
+                            LEGACY_SERVICE_IMPL_NAME_PROPERTY),
+                    clazz, platformConfig);
+        }
+
+        private static Optional<String> getOptionalFirstProperty(ModuleConfig moduleConfig,
+                List<String> propertyNames) {
+            return propertyNames.stream().map(moduleConfig::getOptionalStringProperty)
+                    .filter(Optional::isPresent).map(Optional::get).findFirst();
```

Comment:
Coding style: To increase the readability, we should write 1 stream function per line

Reply:
ok